src/main/java/seedu/address/ui/HelpWindow.java

```diff
@@ -15,88 +25,150 @@
+
+            // replace GitHub/Jekyll emoji shortcodes with Unicode
+            md = md.replace(":bulb:", "💡")
+                    .replace(":information_source:", "ℹ️")
+                    .replace(":exclamation:", "⚠️");
+
+
+            // 2) Flexmark config with TOC + heading anchors + tables
+            MutableDataSet opts = new MutableDataSet()
+                    .set(Parser.EXTENSIONS, java.util.List.of(
+                            TablesExtension.create(),
+                            TocExtension.create()
+                    ))
+                    // generate id="" attributes on headings so the TOC can link to them
+                    .set(HtmlRenderer.GENERATE_HEADER_ID, true)
+                    .set(HtmlRenderer.RENDER_HEADER_ID, true)
+                    // optional: GitHub-like id generation
+                    .set(HtmlRenderer.HEADER_ID_GENERATOR_RESOLVE_DUPES, true)
+                    .set(TocExtension.LIST_CLASS, "toc")
+                    .set(TocExtension.LEVELS, 255);
+
+            Parser parser = Parser.builder(opts).build();
+            HtmlRenderer renderer = HtmlRenderer.builder(opts).build();
+
+            Node document = parser.parse(md);
+            String bodyHtml = renderer.render(document);
+
+            // 3) Base href so relative images resolve inside the JAR
+            String baseHref = computeDocsBaseHref();
+
+            String css = """
+                        body{font-family:-apple-system,Segoe UI,Roboto,Arial,sans-serif;padding:16px;line-height:1.6}
+                        h1,h2,h3{border-bottom:1px solid #ddd;padding-bottom:.25em}
+                        img{max-width:100%;height:auto}
+                        code, pre{font-family:ui-monospace,SFMono-Regular,Menlo,Consolas,monospace}
+                        .toc{margin:0 0 1rem 0;padding-left:1.25rem}
+                        /* Minimal Bootstrap-like alerts */
+                        .alert{border-radius:6px;padding:.75rem 1rem;margin:1rem 0;border:1px solid #d0d7de;
+                        background:#f6f8fa}
+                        .alert-info{border-color:#54aeff;background:#ddf4ff}
+                        .alert-primary{border-color:#9b8eff;background:#f4f1ff}
+                        .alert-warning{border-color:#c69026;background:#fff8c5}
+                    """;
+
+            String html = "<!doctype html><html><head><meta charset='UTF-8'>"
+                    + (baseHref != null ? "<base href='" + baseHref + "'>" : "")
+                    + "<style>" + css + "</style></head><body>"
+                    + bodyHtml
+                    + "</body></html>";
+
+            webView.getEngine().loadContent(html);
+        } catch (Exception e) {
+            webView.getEngine().loadContent(
+                    "<h2>Failed to load UserGuide</h2><pre>" + e.getMessage() + "</pre>");
+        }
+    }
+
+    private String readUserGuideMarkdown() throws Exception {
```

Comment:
Nit: Could the exception specified be more specific?


```suggestion
    private String readUserGuideMarkdown() throws IOException {
```

See: https://stackoverflow.com/a/8082576

> ...more generally you should not `throws Exception`.